src/pubtools/_pulp/tasks/delete.py

```diff
@@ -564,21 +566,29 @@ def map_to_repo(self, units, repos, unit_attr):
         repo_map = {}
         unit_map = {}
         repos = sorted(repos)
-
         for unit in sorted(units):
             unit_name = getattr(unit, unit_attr)
             unit_map.setdefault(unit_name, RemoveUnitItem(unit=unit, repos=[]))
-            for repo in repos:
-                if repo not in unit.repository_memberships:
-                    LOG.warning(
-                        "%s is not present in %s",
-                        unit_name,
-                        repo,
-                    )
-                else:
+            if "*" in repos:
```

Comment:
Minor: not really liking the fact that this goes into two separate branches.
It means, for example, that if someone later refactors the two lines starting from `repo_map.setdefault(repo, []).append(unit)` they will need to refactor both branches or will possibly waste time forgetting to update one of them.

It seems like (in psuedocode) it could all be handled like this:

```
# note: repos probably needs to be copied before this, if the caller doesn't expect
# it to be mutated.
if repos contains the wildcard:
  update repos, replacing the wildcard with all repos containing the unit
process all repos
```

Right now the "process all repos" bit appears in the code twice.